Curvature/HTTPRequest/HTTPRequest.swift

```diff
@@ -32,6 +32,7 @@ public struct HTTPRequest {
 
     public var parameters: [String: String] = [:]
     public var data: [String: Any] = [:]
+    public var context: [String: Any] = [:]
```

Comment:
I think the `data` variable should represent a deserialised dictionary of data parsed from the request. This would make sense since Camarilla overwrites the data property. (https://github.com/Zewo/Camarilla/blob/master/Camarilla/JSONParserMiddleware.swift)
This might not be a problem, but the order of middleware becomes a potential point of confusion - hence a separate variable that cannot be confused with say, body data.